tests/src/test/scala/com/nvidia/spark/rapids/GpuMultiFileReaderSuite.scala

```diff
@@ -74,4 +78,106 @@ class GpuMultiFileReaderSuite extends AnyFunSuite {
       assertResult(false)(multiFileReader.next())
     }
   }
+
+  def generateRandomString(length: Int): String = {
+    Iterator.fill(length)(Random.alphanumeric.head).mkString
+  }
+
+  def sqlExprToGenerateRandomString(length: Int, partitions: Int): String = {
+    val randomString = generateRandomString(length)
+    s"substring('$randomString', 1, floor(rand() * $partitions) + 1)"
+  }
+
+  def writeAndReadParquet(df: DataFrame, spark: SparkSession, outputFile: File,
+      partCols:String*): DataFrame = {
+    spark.conf.set("spark.rapids.sql.enabled", "false")
+    df.write.partitionBy(partCols:_*).parquet(outputFile.getCanonicalPath)
+
+    spark.conf.set("spark.rapids.sql.enabled", "true")
+    spark.conf.set("spark.rapids.cudfColumnSizeLimit", 200000)
+    spark.read.parquet(outputFile.getCanonicalPath)
+  }
+
+  test("WIP: test column size exceeding cuDF limit" +
+    " - single partition column, single partition value") {
+    val numRows = 10000
+    withTempPath { file =>
+      withGpuSparkSession(spark => {
+        val df = spark.range(numRows)
+          .withColumn("partCol", lit(generateRandomString(150)))
+        val resultAns = writeAndReadParquet(df, spark, file, "partCol").collect()
+        assert(resultAns.length == numRows)
```

Comment:
This test doesn't actually verify the values of the rows are correct.  This probably should run with the CPU reading the same Parquet data and verify the output is correct, not just a row count.  There can be so many ways the code is broken but still get the row count right.

I think this approach to testing would be better handled in the pyspark tests, as there are nice data generators there and I don't see anything that would preclude it.  In general, we prefer to write tests in pyspark rather than Scala.

Reply:
In pyspark tests, will add verification of data.